sdk/download.go

```diff
@@ -50,15 +52,38 @@ func TarballURL(version string) string {
 	return u.String()
 }
 
-func DownloadFile(file, url string) error {
-	plog.Infof("Downloading %s to %s", url, file)
+func DownloadFile(file, fileURL string, client *http.Client) error {
+	plog.Infof("Downloading %s to %s", fileURL, file)
 
+	// handle bucket urls by using api to get media link
+	parseURL, err := url.Parse(fileURL)
+	if err != nil {
+		return err
+	}
+	if parseURL.Scheme == "gs" {
+		if client == nil {
+			return fmt.Errorf("DownloadFile requires non-nil client with gs:// links")
```

Comment:
The error message makes it sound like a programming error which should be a panic but since the URL is probably user provided I'd switch the if statements and only try to handle gs if client is not nil. Or you can just use the default if it is possible to use the api auth-free for public buckets/objects.